### ISSUE TITLE
AttributeError: type object 'InvokeOptions' has no attribute 'merge'

### ISSUE THREAD

**Issue by idrissneumann** [pulumi/pulumi-openstack#240]
### What happened?

Hi.

Seems there's a missing `merge` attribute on the `InvokeOptions`.

Here's our code:

```python
def create_pulumi_program():
    registry = compute.get_flavor(name=hashed_registry_name)
    pulumi.export("type", registry.flavor_name)
            
stack = auto.create_or_select_stack(stack_name=hashed_registry_name, project_name=user_email, program=create_pulumi_program)
        
up_res = stack.up()
```

And the result:

```shell
Diagnostics:
  pulumi:pulumi:Stack (idriss.neumann@comwork.io-test-reg-ovh-ml1isq):
    error: python inline source runtime error: type object 'InvokeOptions' has no attribute 'merge'
    Traceback (most recent call last):
      File "/usr/local/lib/python3.9/site-packages/pulumi/automation/_server.py", line 69, in Run
        loop.run_until_complete(run_in_stack(self.program))
      File "/usr/local/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
        return future.result()
      File "/usr/local/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 126, in run_in_stack
        await run_pulumi_func(lambda: Stack(func))
      File "/usr/local/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 49, in run_pulumi_func
        func()
      File "/usr/local/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 126, in <lambda>
        await run_pulumi_func(lambda: Stack(func))
      File "/usr/local/lib/python3.9/site-packages/pulumi/runtime/stack.py", line 149, in __init__
        func()
      File "/app/src/strategies/OvhStrategry.py", line 65, in create_pulumi_program
        registry = compute.get_flavor(name=hashed_registry_name)
      File "/usr/local/lib/python3.9/site-packages/pulumi_openstack/compute/get_flavor.py", line 212, in get_flavor
        opts = pulumi.InvokeOptions.merge(_utilities.get_invoke_opts_defaults(), opts)
    AttributeError: type object 'InvokeOptions' has no attribute 'merge'
    error: an unhandled error occurred: python inline source runtime error: type object 'InvokeOptions' has no attribute 'merge'
    Traceback (most recent call last):
```

Thanks for your help

### Expected Behavior

`InvokeOptions` should have a `merge` method.

### Steps to reproduce

Just try this piece of code:

```python
def create_pulumi_program():
    registry = compute.get_flavor(name=hashed_registry_name)
    pulumi.export("type", registry.flavor_name)
            
stack = auto.create_or_select_stack(stack_name=hashed_registry_name, project_name=user_email, program=create_pulumi_program)
        
up_res = stack.up()
```

### Output of `pulumi about`

```shell
$ docker exec -it comwork_cloud_api_ppd bash
root@af62a64b35aa:/app# pulumi about
CLI          
Version      3.55.0
Go Version   go1.20
Go Compiler  gc

Host     
OS       debian
Version  11.6
Arch     x86_64

Backend        
Name           af62a64b35aa
URL            s3://tf-pulumi-state?endpoint=s3.fr-par.scw.cloud
User           root
Organizations  

Pulumi locates its logs in /tmp by default
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Hey @idrissneumann, can you try updating the Pulumi SDK for Python? I think this is resolved in the `pulumi` dependency version 3.35.0 and newer.

**Comment by AaronFriel** [comment]
We're confident this is resolved when using Pulumi 3.35 & the 3.35 Python SDK or newer, please let us know if not by creating a new issue for us to track.
